### PR TITLE
Card html/js

### PR DESCRIPTION
**Pull request by jwolfcrafter**
Both the html and javascript for the card modal

### CODE REVIEW

File: src/public/js/cardModal.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            let requestData = {
+                method: "POST", //I think?
+                url: `TODO/${boardId}/${cardId}` //TODO: Insert route to get info
+            };
+            //make the call to ajax to populate
+            $.ajax(requestData).then(function(responseMessage) {
+                let cardModal = $(responseMessage);
+                placeModalHere.empty(); //handle a modal being there before
+                placeModalHere.append(cardModal); //insert new modal
+                setUpErrorChecking();
```
Comment (mpolimen) [inline src/public/js/cardModal.js:23]: Typically, declare functions above where they are used

File: src/views/partials/card.handlebars [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{{#each card}}
+<div class="card-modal" hidden>
+    <button class="close-button" id="closeModal">Close Modal</button>
+    <form id="updateCard" action="/board/card/{{boardId}}/{{_id}}" method="POST">
+        <h1>{{name}}</h1>
+        <h2>Members</h2>
+        {{#each members}}
+        {{#if assigned}}
+        <h3 class="assigned">{{initials}}</h3>
```
Comment (mpolimen) [inline src/views/partials/card.handlebars:9]: Not sure if this makes sense, what are initials in this case? All the initials that are assigned? Comma-separated or by space?

**Comment by mpolimen** [comment]
This would still need the CSS to position it like a modal, right?

File: src/views/partials/card.handlebars (hunk 2 above)
Comment (jwolfcrafter) [reply src/views/partials/card.handlebars:9]: For each of the members it checks if they're assigned or not, members[x].initials is their initials like JP or MP like we would display in the circle, I just imagined we would display something different for members who are assigned to the card and members who aren't yet assigned

**Comment by jwolfcrafter** [comment]
Yes, all of the classes that are put in are essentially guides for the person writing the css, so everything including the modal would still need to have css done

File: src/public/js/cardModal.js (hunk 1 above)
Comment (jwolfcrafter) [reply src/public/js/cardModal.js:23]: I'm not sure what you meant by this, could you elaborate?

Comment (mpolimen) [reply src/public/js/cardModal.js:23]: You call setUpErrorChecking() on line 23, but it's not declared until after line 23. In some languages, this wouldn't compile. In general, it's good code practice to move function declarations above their respective function calls

This is no reason not to merge, just mentioning preferred practice

**Comment by mpolimen** [comment]
This gets a pass by me

File: src/views/partials/card.handlebars [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{{#each card}}
+<div class="card-modal" hidden>
+    <button class="close-button" id="closeModal">Close Modal</button>
+    <form id="updateCard" action="/board/card/{{boardId}}/{{_id}}" method="POST">
```
Comment (roccopolimen) [inline src/views/partials/card.handlebars:4]: using id in a loop is gonna make multiple things with the same id

File: src/views/partials/card.handlebars [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        <h1>{{name}}</h1>
+        <h2>Members</h2>
+        {{#each members}}
+        {{#if assigned}}
```
Comment (roccopolimen) [inline src/views/partials/card.handlebars:8]: the assigned should be a checkbox input so you can select and unselect people and then submit the form to change who is assigned

File: src/views/partials/card.handlebars [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            Position
+            <select id="position">
+                {{#each position}}
+                <option value="{{position}}">{{position}}</option>
+                {{/each}}
+            </select>
+        </label>
+        <label>
+            Due Date
+            <input type="date" id="dueDate" value="{{dueDate.date}}">
```
Comment (roccopolimen) [inline src/views/partials/card.handlebars:39]: type="date" does not do anything with the time, should have a second input for the time as well

File: src/public/js/cardModal.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
+(function($) {
+    //this file should be imported into the board handlebars file
+    //this variable should be the <div> that the modal is going to be inserted in to
+    let placeModalHere = $("#placeModalHere");
```
Comment (roccopolimen) [inline src/public/js/cardModal.js:4]: look at my board html and let me know if what I have does not work with how you plan for this to work, I can change it to add another div with some id="modal-container" or something like that

**Review by roccopolimen — COMMENTED** [review]
Few things, also might be helpful for you and i to get on a call sometime this week to discuss how my board page and this card modal need to interact so it works properly. LMK

File: src/views/partials/card.handlebars (hunk 3 above)
Comment (jwolfcrafter) [reply src/views/partials/card.handlebars:4]: your route should only ever be giving me 1 card, this is just so I dont have to type card._id, card.members, card.name, etc.

Comment (roccopolimen) [reply src/views/partials/card.handlebars:4]: okay dope

File: src/public/js/cardModal.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
```
Comment (roccopolimen) [inline src/public/js/cardModal.js:4]: We'll get rid of this section of the code when we Frankenstein this functionality into board-page

**Review by roccopolimen — COMMENTED** [review]
Looks pree neat

**Pull request merged**